### PR TITLE
Add new `Lint/NoopRescue` cop

### PR DESCRIPTION
**Pull request by ydah**
This PR is add new `Lint/NoopRescue` cop.
This cop check for suppress or ignore checked exception.

Suppressing or ignoring exceptions hides the problem, makes it invisible, and makes it behave as if it were not a problem.
The test code becomes less reliable and the possibility of problems increases.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [ ] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Review by dvandersluis — CHANGES_REQUESTED** [review]
If we're going to add this, it needs to be way more permissive than it currently is.

First off `foo rescue nil` is a very common pattern. Moreso, there are a lot of returned values in a rescue that are not `send` nodes but are not no-ops. For instance:

```ruby
def get_api_value
  MyAPI.request(foo, bar)
rescue APIError
  nil
end
```

```ruby
def resource_available?
  Resource.connect
rescue Connection::TimedOut
  false
end
```

These are completely valid, non-noop responses that will fail this cop.

IMO, this cop should maybe just catch rescues with no body but leave everything else as allowed.

File: config/default.yml [hunk 1]
```diff
@@ -1994,6 +1994,11 @@ Lint/NonLocalExitFromIterator:
   Enabled: true
   VersionAdded: '0.30'
 
+Lint/NoopRescue:
+  Description: 'Check for suppress or ignore checked exception.'
```
Comment (dvandersluis) [inline config/default.yml:1998]:
```suggestion
  Description: 'Check for suppressed or ignored rescued exceptions.'
```

File: lib/rubocop/cop/lint/noop_rescue.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Check for suppress or ignore checked exception.
```
Comment (dvandersluis) [inline lib/rubocop/cop/lint/noop_rescue.rb:6]: Same here.

File: lib/rubocop/cop/lint/noop_rescue.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      #
+      #   # good
+      #   def foo
+      #     do_something
+      #   rescue StandardError => e
+      #     do_something
+      #   end
+      #
+      class NoopRescue < Base
+        MSG = "Don't suppress or ignore checked exception."
```
Comment (dvandersluis) [inline lib/rubocop/cop/lint/noop_rescue.rb:30]:
```suggestion
        MSG = "Don't suppress or ignore rescued exceptions."
```

File: spec/rubocop/cop/lint/noop_rescue_spec.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      expect_no_offenses(<<~RUBY)
+        begin
+          do_something
+        rescue StandardError
+          do_something
+        end
+      RUBY
+    end
+
+    it 'does not registers an offense when catching multiple exceptions' do
```
Comment (dvandersluis) [inline spec/rubocop/cop/lint/noop_rescue_spec.rb:70]:
```suggestion
    it 'does not register an offense when catching multiple exceptions' do
```

File: spec/rubocop/cop/lint/noop_rescue_spec.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    it 'does not register an offense when inside the method definition' do
+      expect_no_offenses(<<~RUBY)
+        def foo
+        rescue
+          do_something
+        end
+      RUBY
+    end
+
+    it 'does not registers an offense when begin-rescue' do
```
Comment (dvandersluis) [inline spec/rubocop/cop/lint/noop_rescue_spec.rb:60]:
```suggestion
    it 'does not register an offense with begin-rescue' do
```

**Comment by dvandersluis** [comment]
Actually, see `Lint/SuppressedExceptions`. We already have a cop to do this, I'm afraid 😅

**Comment by ydah** [comment]
If I thought it was not pointed out, I could detect it by setting the default: false option to true. (I hadn't noticed. Thank you so much.)

```yaml
Lint/SuppressedException:
  AllowComments: false
  AllowNil: false
````

**Pull request closed without merging**